tests/scripts/github-action-helper.sh

```diff
@@ -738,6 +762,59 @@ function install_minikube_with_none_driver() {
   minikube start --kubernetes-version="$1" --driver=none --memory 6g --cpus=2 --addons ingress --cni=calico
 }
 
+function toolbox() {
+  kubectl -n rook-ceph exec -it "$(kubectl -n rook-ceph get pod -l "app=rook-ceph-tools" -o jsonpath='{.items[0].metadata.name}')" -- "$@"
+}
+
+function ceph() {
+  toolbox ceph "$@"
+}
+
+function rbd() {
+  toolbox rbd "$@"
+}
+
+function radosgw-admin() {
+  toolbox radosgw-admin "$@"
+}
+
+function test_object_with_cephblockpools_extra_pools() {
+  expected_pools=(
+    .mgr
+    .rgw.root
+    object-with-cephblockpools.rgw.control
+    object-with-cephblockpools.rgw.meta
+    object-with-cephblockpools.rgw.log
+    object-with-cephblockpools.rgw.buckets.index
+    object-with-cephblockpools.rgw.buckets.non-ec
+    object-with-cephblockpools.rgw.otp
+    object-with-cephblockpools.rgw.buckets.data
+  )
+
+  output=$(ceph osd pool ls)
+  readarray -t live_pools < <(printf '%s' "$output")
+
+  errors=0
+  for l in "${live_pools[@]}"; do
+    found=false
+    for e in "${expected_pools[@]}"; do
+      if [[ "$l" == "$e" ]]; then
+        found=true
+        break
+      fi
+    done
+    if [[ "$found" == false ]]; then
+      echo "Live pool $l is not an expected pool"
+      errors=$((errors+1))
+    fi
+  done
+
+  if [[ $errors -gt 0 ]]; then
+    echo "Found $errors errors"
+    exit $errors
+  fi
+}
```

Comment:
One of the things that I have come to realize over time is that stuffing a boatload of stuff into one script is actually really hard to grok when moving between unit test (GH Action def) and script. I'd like to propose moving this check to a separate script. I think it'll do well as a standalone since it doesn't hinge on other functions in this script, but let me know what you think @jhoblitt . As a newcomer, are you also finding the same challenge, and do you think using a dedicated script for this will be an improvement?

Reply:
Line 794 is using `ceph()` which is cherry-picked from https://github.com/rook/rook/pull/14789.  Breaking this function out into its own file would require changing that line to something along the lines of:

```bash
  output=$("${REPO_DIR}/tests/scripts/github-action-helper.sh" ceph osd pool ls)
```

I'm more or less ambivalent about trying to decompose `github-action-helper.sh`.  I find the entry point behavior fairly obvious. I suspect that breaking it up could cause more maintenance burden trying to combat duplicated functions.  I do think it would be good to have an overall plan for how to structure "acceptance" tests before starting deep refactoring.

Reply:
Oh, missed that! Thanks!

Reply:
Is it acceptable as-is or do you want me to split it out?

Reply:
Good as-is! Thanks